src/server/node.rs

```diff
@@ -192,16 +192,43 @@ impl<C> Node<C>
               peer_id,
               region_id);
 
-        let region = try!(store::bootstrap_region(engine, store_id, region_id, peer_id));
+        let region = try!(store::prepare_bootstrap(engine, store_id, region_id, peer_id));
         Ok(region)
     }
 
+    fn check_prepare_bootstrap_cluster(&self, engine: &DB) -> Result<()> {
+        let res = try!(engine.get_msg::<metapb::Region>(&keys::prepare_bootstrap_key()));
+        if res.is_none() {
+            return Ok(());
+        }
+
+        let region_id = res.unwrap().get_id();
+        for _ in 0..MAX_CHECK_CLUSTER_BOOTSTRAPPED_RETRY_COUNT {
+            match self.pd_client.get_region(b"") {
+                Ok(region) => {
+                    if region.get_id() == region_id {
+                        try!(store::clear_prepare_bootstrap_state(engine));
+                    } else {
+                        try!(store::clear_prepare_bootstrap(engine, region_id));
+                    }
+                    return Ok(());
+                }
+
+                Err(e) => {
+                    warn!("check cluster prepare bootstrapped failed: {:?}", e);
+                }
+            }
+            thread::sleep(Duration::from_secs(CHECK_CLUSTER_BOOTSTRAPPED_RETRY_SECONDS));
+        }
+        Err(box_err!("check cluster prepare bootstrapped failed"))
+    }
+
     fn bootstrap_cluster(&mut self, engine: &DB, region: metapb::Region) -> Result<()> {
         let region_id = region.get_id();
         match self.pd_client.bootstrap_cluster(self.store.clone(), region) {
             Err(PdError::ClusterBootstrapped(_)) => {
                 error!("cluster {} is already bootstrapped", self.cluster_id);
-                try!(store::clear_region(engine, region_id));
+                try!(store::clear_prepare_bootstrap(engine, region_id));
```

Comment:
I guess you want to `clear_prepare_bootstrap_state()` if pd returns Ok() at L237.